### ISSUE TITLE
Localization remaps don't work for themes in exported project

### ISSUE THREAD

**Issue by conest** [godotengine/godot#97263]
### Tested versions

4.3 stable

### System information

Godot v4.3.stable - macOS 14.5.0 - Vulkan (Forward+) - integrated Apple M2 Pro - Apple M2 Pro (10 Threads)

### Issue description

I attempted to use localization remaps to switch between different themes(.tres files) to suit various languages and apply different font styles. Everything worked fine during testing in the editor, but after exporting, the remap didn’t function as expected — the label’s font and size didn’t change.
Maybe related to this one: [Changing the locale breaks remapped fonts](https://github.com/godotengine/godot/issues/80130) 

### Steps to reproduce

1. Run the root.tscn.
2. Press 2 buttons to change the locale and check the font on the label.
3. Export project.
4. Run the exported project and follow the step 2 again.

### Minimal reproduction project (MRP)

[Archive.zip](https://github.com/user-attachments/files/17082499/Archive.zip)

**Comment by Vovkiv** [comment]
Exported game output this into terminal:
```
WARNING: Translation remap 'res://font_en.tres' does not exist. Falling back to 'res://font_en.tres'.
     at: _path_remap (core/io/resource_loader.cpp:1197)
WARNING: Translation remap 'res://font_ja.tres' does not exist. Falling back to 'res://font_en.tres'.
     at: _path_remap (core/io/resource_loader.cpp:1197)
```

Seems that this 2 resources doesn't get included in exported build?

**Comment by conest** [comment]
> Exported game output this into terminal:
> 
> ```
> WARNING: Translation remap 'res://font_en.tres' does not exist. Falling back to 'res://font_en.tres'.
>      at: _path_remap (core/io/resource_loader.cpp:1197)
> WARNING: Translation remap 'res://font_ja.tres' does not exist. Falling back to 'res://font_en.tres'.
>      at: _path_remap (core/io/resource_loader.cpp:1197)
> ```
> 
> Seems that this 2 resources doesn't get included in exported build?

Yes, I tried exporting them explicitly, but it still didn’t work.
<img width="398" alt="Screenshot 2024-09-21 at 14 28 46" src="https://github.com/user-attachments/assets/83578a4c-3b3b-4675-ba50-231716a488ee">

**Comment by Vovkiv** [comment]
Adding new labels and applying that themes to them - works, so resources definitely included in exported build.
![зображення](https://github.com/user-attachments/assets/e6bcd9ad-52ae-400b-b01f-fe79b16899fc)

**Comment by conest** [comment]
> Adding new labels and applying that themes to them - works, so resources definitely included in exported build. ![зображення](https://private-user-images.githubusercontent.com/54743395/369648595-e6bcd9ad-52ae-400b-b01f-fe79b16899fc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY5Njc5MDAsIm5iZiI6MTcyNjk2NzYwMCwicGF0aCI6Ii81NDc0MzM5NS8zNjk2NDg1OTUtZTZiY2Q5YWQtNTJhZS00MDBiLWIwMWYtZmU3OWIxNjg5OWZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIyVDAxMTMyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdhMjE5MjZjMDgwMGI5YWE5ZjFmODEyMTc0MTU4MWRmYjVjZDgyN2ViNTY5NTAwM2JjZTI1OWRkODVkZmY3NTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.8hx1QXntRbwm4VjL7vONLxt07XdFomRlPR4kvgN0nu0)

My tests reached the same conclusion. It seems that the remaps resources are not shared with the regular resources, or not properly linked for some reason.

**Comment by geekley** [comment]
I'm having this same issue and error message, but on a custom resource, unrelated to themes, and only in Web exports.
I tried even using metadata to reference a resource that points to all locale variations, but no luck.
I guess I'll have to avoid the localization remap system entirely and use my own code for this... 😞

**Comment by ebeem** [comment]
I am facing the same issue, I think it's related to #84791
After checking my .pck file, I noticed that the extensions of remaps is `.remap`

For example, my file `main-menu-theme.tres` will become `main-menu-theme.tres.remap` rather than `main-menu-theme.tres.import`
So how to fix this? I think we need to skip also the extension `.remap`

I will see if I can test this, my project uses mono, it's much harder to compile, but I will try

```cpp
	if (!FileAccess::exists(new_path + ".import") && !FileAccess::exists(new_path + ".remap") && !FileAccess::exists(new_path)) {
			WARN_PRINT(vformat("Translation remap '%s' does not exist. Falling back to '%s'.", new_path, p_path));
			new_path = p_path;
		}
```

**Comment by ebeem** [comment]
I can confirm that the inclusion of ".remap" in file exists solves the problem
Will create a PR for this

**Comment by ebeem** [comment]
Created the PR #103838
You can test it if you would like to, but a export template will have to be built for each target. Which is a lot of effort.
